### PR TITLE
build(docker): add dockerfile

### PR DESCRIPTION
**Pull request by codeswhite**
Resolving #8

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+FROM node:20-alpine AS build
+
+RUN npm install -g pnpm
+
+WORKDIR /app
+COPY . .
+
+# ? Consider --frozen-lockfile
+RUN pnpm install
```
Comment (async3619) [inline Dockerfile]: would be nice to have `--frozen-lockfile` imo

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+# ? Add checks?
+
+# ? Consider --prod
+RUN pnpm run build
```
Comment (async3619) [inline Dockerfile]: afaik, we can make production artifact by simply using `pnpm run build` so we don't need to add --prod flag

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+
+# Stage 2: Serve
+
+FROM caddy:alpine
+
+COPY --from=build /app/dist /usr/share/caddy
+
+EXPOSE 80
+CMD ["caddy", "file-server", "--root", "/usr/share/caddy"]
```
Comment (async3619) [inline Dockerfile:24]: just curiosity, Is caddy good for simple file serving?
I've never used caddy before

**Review by async3619 — CHANGES_REQUESTED** [review]

**Comment by async3619** [comment]
and you should add `.dockerignore` and add `node_modules` to make docker image having correct dependency binaries since rollup have native modules

File: Dockerfile (hunk 3 above)
Comment (codeswhite) [reply Dockerfile:24]: Its a strong web server, relatively modern and fast, a common alternative for Nginx and Apache.
I have been using it in some places, love it so far.

Regarding the use case, as you see it has a built-in option to serve files from a folder, I believe its a good solution  overall.

**Comment by codeswhite** [comment]
Regarding the Rollup nuance,
Ive added a `.dockerignore` file

Although im pretty sure the `pnpm install` inside the dockerfile takes care of installing the correct native binaries in case they are incompatible

**Review by async3619 — APPROVED** [review]

**Comment by codeswhite** [comment]
https://www.youtube.com/watch?v=N5PAU-vYrN8
One of my favorite youtubers did a very nice benchmarking video comparing Nginx and Caddy.

To be honest I didnt expect Caddy to perform this bad compared to Nginx.
But besides the performance its much more simple out of the box,
which is an advantage against Nginx and Apache.

**Pull request merged**